napari/_qt/qt_plugin_list.py

```diff
@@ -0,0 +1,199 @@
+"""Provides a QtPluginSorter that allows the user to change plugin call order.
+"""
+from qtpy.QtCore import Qt, Signal
+from qtpy.QtWidgets import (
+    QComboBox,
+    QDialog,
+    QListWidget,
+    QListWidgetItem,
+    QLabel,
+    QSizePolicy,
+    QVBoxLayout,
+    QWidget,
+)
+from ..plugins import plugin_manager as napari_plugin_manager
+from ..plugins.utils import permute_hook_implementations, HookOrderType
+from pluggy.manager import PluginManager, _HookCaller
+from typing import Optional
+
+
+class QtHookImplItem(QListWidgetItem):
+    """Item in a QtHookImplListWidget."""
+
+    def __init__(self, hookimpl, parent=None):
+        self.plugin_name = hookimpl.plugin_name
+        if self.plugin_name == 'builtins':
+            self.plugin_name = 'napari built-in'
+        super().__init__(self.plugin_name, parent)
+        self.hookimpl = hookimpl
+        self.setBackground(Qt.black)
+
+
+class QtHookImplListWidget(QListWidget):
+    """A QListWidget that allows sorting of plugin hooks."""
+
+    order_changed = Signal(list)  # emitted when the user changes the order.
+
+    def __init__(
+        self,
+        hook: Optional[_HookCaller] = None,
+        parent: Optional[QWidget] = None,
+    ):
+        """A ListWidget that shows all of the hook implementations for a spec.
+
+        Usually instantiated by QtPluginSorter.
+
+        Parameters
+        ----------
+        hook : pluggy.manager._HookCaller, optional
+            A pluggy HookCaller to show implementations for. by default None
+            (i.e. no hooks shown)
+        parent : QWidget, optional
+            Optional parent widget, by default None
+        """
+        super().__init__(parent)
+        self.setDefaultDropAction(Qt.MoveAction)
+        self.setDragEnabled(True)
+        self.setDragDropMode(self.InternalMove)
+        self.setAcceptDrops(True)
+        self.setSpacing(1)
+        self.setMinimumHeight(1)
+        self.setSizePolicy(
+            QSizePolicy.MinimumExpanding, QSizePolicy.MinimumExpanding
+        )
+        self.order_changed.connect(self.permute_hook)
+        self.set_hook(hook)
+
+    def set_hook(self, hook: Optional[_HookCaller]) -> None:
+        """Set the list widget to show hook implementations for ``hook``.
+
+        Parameters
+        ----------
+        hook : _HookCaller, optional
+            A pluggy HookCaller to show implementations for. by default None
+            (i.e. no hooks shown)
+        """
+        self.clear()
+        self.hook = hook
+        if not hook:
+            return
+        for hookimpl in reversed(hook.get_hookimpls()):
+            self.addItem(QtHookImplItem(hookimpl))
```

Comment:
These are reversed because `hook.get_hookimpls()` actually returns hook implementations in an order _reversed_ from their call order.  And I wanted the GUI to show it in call order, which I thought was more intuitive.  [Here is the line in pluggy](https://github.com/pytest-dev/pluggy/blob/6e8b6d4e8e0519e50af05b4e4b90b18b924bcbcc/src/pluggy/callers.py#L169) where the list of hook implementations (the output of `get_hookimpls()`) is reversed before calling them.

Reply:
Missed this originally. Blows my mind. LOL Thanks for adding the code comments! I expect they will come in handy in the future. =)